### PR TITLE
[cart] document instrumentation

### PR DESCRIPTION
**Pull request by puckpuck**
Adds documentation for cart service, focused on instrumentation code.

### CODE REVIEW

**Review by mic-max — APPROVED** [review]

**Comment by cartersocha** [comment]
@puckpuck what are your thoughts on adding language to the file names / titles?

I feel like a user would prefer to see a language and go directly there rather than consult our architecture for direction

**Comment by mviitane** [comment]
> I feel like a user would prefer to see a language and go directly there rather than consult our architecture for direction

Seems like a great idea! The language could also be mentioned in the top section of each doc, for example, next to the source code link.

**Review by mviitane — APPROVED** [review]
LGTM

**Comment by puckpuck** [comment]
> @puckpuck what are your thoughts on adding language to the file names / titles?
> 
> I feel like a user would prefer to see a language and go directly there rather than consult our architecture for direction

I think this is a great idea.

`<servicename>_<language>.md` make sense for the file names?

I will also add it to the title for each one also.  I need to make a few small changes to this PR so will do that here, and submit another one to clean up all the other docs we wrote thus far.

**Comment by cartersocha** [comment]
Awesome!

File: docs/services/cartservice.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+This service maintains items placed in the shopping cart by users. It interacts
+with a Redis caching service for fast access to shopping cart data.
+
+[Cart service source](../../src/cartservice/)
+
+## SDK Initialization
+
+he OpenTelemetry dotnet SDK should be initialized in your application's
+`Startup.cs` as part of the `ConfigureServices` function. When initializing,
+optionally specify which auto-instrumentation libraries to leverage. The SDK is
```
Comment (cijothomas) [inline docs/services/cartservice.md:12]:
```suggestion
optionally specify which instrumentation libraries to leverage. The SDK is
```

Comment (cijothomas) [reply docs/services/cartservice.md:12]: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/glossary.md#instrumentation-library

File: docs/services/cartservice.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+initialized using a builder pattern, where you add each auto-instrumentation
```
Comment (cijothomas) [inline docs/services/cartservice.md:13]:
```suggestion
initialized using a builder pattern, where you add each instrumentation library
```

**Review by cijothomas — APPROVED** [review]
LGTM. Please do replace auto-instrumentation with instrumentation library.

**Comment by cartersocha** [comment]
> LGTM. Please do replace auto-instrumentation with instrumentation library.

We'll want to update this in all service docs but can wait until we finish adding them

**Pull request merged**